tests/unit/dataactvalidator/test_b12_object_class_program_activity_1.py

```diff
@@ -0,0 +1,94 @@
+from tests.unit.dataactcore.factories.staging import ObjectClassProgramActivityFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+
+_FILE = 'b12_object_class_program_activity_1'
+
+
+def test_column_headers(database):
+    expected_subset = {'row_number', 'by_direct_reimbursable_fun'}
+    actual = set(query_columns(_FILE, database))
+    assert (actual & expected_subset) == expected_subset
+
+
+def test_success(database):
+    """ Test for USSGL 48XX & 49XX (except 487X & 497X) if any one is provided then
+    by_direct_reimbursable_fun is not empty """
+
+    op = ObjectClassProgramActivityFactory()
+    assert number_of_errors(_FILE, database, models=[op]) == 0
+
+    op = ObjectClassProgramActivityFactory(object_class=1234)
+    assert number_of_errors(_FILE, database, models=[op]) == 0
+
+
+def test_failure(database):
+    """ Test for USSGL 48XX & 49XX (except 487X & 497X) if any one is provided then
+    by_direct_reimbursable_fun is empty """
+
+    op = ObjectClassProgramActivityFactory(ussgl480100_undelivered_or_fyb=None, by_direct_reimbursable_fun=None,
+        object_class = 123)
+    assert number_of_errors(_FILE, database, models=[op]) == 1
```

Comment:
Similar question as in `b12_award_financial_1` -- do we need to set all of the fields except one to `None`?